students/EricAdams/session03/string_formatting_lab/string_formatting_lab.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+""" Write a format string that will take the tuple:
+( 2, 123.4567, 10000, 12345.67)
+
+and produce:
+
+'file_002 :   123.46, 1.00e+04, 1.23e+04'
+"""
+
+
+def format_tuple(seq):
+    print('file_{:0>3d} :   {:.2f}, {:.2e}, {:.2e}'.format(
+        seq[0], seq[1], seq[2], seq[3]))
+
+
+format_tuple((2, 123.4567, 10000, 12345.67))
+
+
+def formatter(in_tuple):
+    """Dynamically Build up format string
+    """
+    form_string = ''
+    for x in in_tuple:
+        form_string = form_string + '{} '
+    temp = form_string.split()
+    s = ','
+    form_string = s.join(temp)
```

Comment:
better to just build it up in a list in the first place, rather than splitting so you can rejoin:

but you got the key idea of dynamically creating a format string -- nice!